fpna/Utils/ExtentionMethods.ipynb

```diff
@@ -70,14 +95,12 @@
             "cell_type": "code",
             "source": [
                 "public static double[] GetCompoundValues(this double initialValue, int shift, int length, double[] values){",
-                "\n    var compound = new double[length];",
-                "\n    ",
-                "\n    for (var i = shift; i < length; i++)",
-                "\n        if (i == shift)",
-                "\n            compound[i] = initialValue;",
-                "\n        else",
-                "\n            compound[i] = compound[i-1] * (1d + values[i - shift]);",
                 "\n    ",
+                "\n    var compound = new double[shift + length];",
+                "\n",
+                "\n    for (var i = 0; i < length; i++)",
+                "\n        compound[shift + i] = initialValue * values.Take(i + 1).Select(x => 1 + x).Aggregate((x, y) => x * y);",
```

Comment:
If the formulas across methods are similar I would immediately extract it. 
You can provide a Func to control the operation (sum or multiplication): 
![image](https://github.com/Systemorph/IFRS17CalculationEngine/assets/103409906/4f8c53e0-b4ad-4603-bee3-d31140821d89)

Reply:
I will add a TODO here